pkgs/os-specific/linux/kernel/linux-xanmod.nix

```diff
@@ -1,9 +1,9 @@
-{ lib, stdenv, buildLinux, fetchFromGitHub, ... } @ args:
+{ lib, stdenv, buildLinux, fetchFromGitHub, realtime ? false, ... } @ args:
 
 let
-  version = "5.13.12";
+  version = if !realtime then "5.13.12" else "5.13.1";
   release = "1";
-  suffix = "xanmod${release}-cacule";
+  suffix = if !realtime then "xanmod${release}-cacule" else "rt${release}-xanmod${release}";
```

Comment:
That is not exactly what I had in mind. I think boost and php are good examples for packages which share common files. By using inputs for different things and supplying them at a common place we can avoid the if else around the file and still re-use common parts.